scalding-parquet/src/main/scala/com/twitter/scalding/parquet/HasColumnProjection.scala

```diff
@@ -1,27 +1,67 @@
 package com.twitter.scalding.parquet
 
+import org.slf4j.LoggerFactory
+
+object HasColumnProjection {
+  val LOG = LoggerFactory.getLogger(this.getClass)
+
+  def requireNoSemiColon(glob: String) = {
+    require(!glob.contains(";"), "A column projection glob cannot contain a ; character")
+  }
+}
+
 trait HasColumnProjection {
+  import com.twitter.scalding.parquet.HasColumnProjection._
 
   /**
+   * Deprecated. Use withColumnProjections, which uses a different glob syntax.
+   *
    * The format for specifying columns is described here:
-   * https://github.com/apache/incubator-parquet-mr/blob/master/parquet_cascading.md#21-projection-pushdown-with-thriftscrooge-records
+   * https://github.com/apache/parquet-mr/blob/3df3372a1ee7b6ea74af89f53a614895b8078609/parquet_cascading.md#2-projection-pushdown
+   * (Note that this link is different from the one below in withColumnProjections)
    *
    * Note that the format described there says that multiple globs can be combined with a ; character.
    * Instead, we use a Set() here and will eventually join the set on the ; character for you.
    */
+  @deprecated(message = "Use withColumnProjections, which uses a different glob syntax", since = "0.15.1")
   def withColumns: Set[String] = Set()
 
-  protected[parquet] final def columnGlobs: Set[ColumnProjectionGlob] = withColumns.map(ColumnProjectionGlob)
+  /**
+   * The format for specifying columns is described here:
+   * https://github.com/apache/parquet-mr/blob/master/parquet_cascading.md#21-projection-pushdown-with-thriftscrooge-records
+   *
+   * Note that the format described there says that multiple globs can be combined with a ; character.
+   * Instead, we use a Set() here and will eventually join the set on the ; character for you.
+   */
+  def withColumnProjections: Set[String] = Set()
```

Comment:
If anyone has a better suggestion to differentiate `withColumns` (deprecated) from `withColumnProjections` (new api) let me know. Could be something shorter like `withColumns2` :(